src/Data/Ratio.purs

```diff
@@ -1,45 +1,64 @@
 module Data.Ratio
-  ( Ratio(Ratio)
+  ( Ratio
+  , reduce
+  , (%)
   , numerator
   , denominator
-  , gcd
   ) where
 
-import Prelude ( class CommutativeRing, class Eq, class EuclideanRing
-               , class Field, class Ring, class Semiring
-               , one, zero, (*), (+), (-)
-               , class Show, show, (<>)
-               )
+import Prelude
 import Prelude ( gcd ) as Prelude
+import Data.Ord (abs, signum)
 
 data Ratio a = Ratio a a
 
 instance showRatio :: Show a => Show (Ratio a) where
-  show (Ratio a b) = "(Ratio " <> show a <> " " <> show b <> ")"
+  show (Ratio a b) = show a <> " % " <> show b
 
-instance semiringRatio :: Semiring a => Semiring (Ratio a) where
+instance eqRatio :: Eq a => Eq (Ratio a) where
+  eq (Ratio a b) (Ratio c d) = a == c && b == d
+
+instance ordRatio :: (Ord a, EuclideanRing a) => Ord (Ratio a) where
+  compare x y =
+    case x - y of
+      Ratio n d ->
+        if n == zero
+          then EQ
+          else case n > zero, d > zero of
+            true, true -> GT
+            false, false -> GT
+            _, _ -> LT
+
+instance semiringRatio :: (Ord a, EuclideanRing a) => Semiring (Ratio a) where
   one = Ratio one one
-  mul (Ratio a b) (Ratio c d) = Ratio (a * c) (b * d)
+  mul (Ratio a b) (Ratio c d) = reduce (a * c) (b * d)
   zero = Ratio zero one
-  add (Ratio a b) (Ratio c d) = Ratio ((a * d) + (b * c)) (b * d)
+  add (Ratio a b) (Ratio c d) = reduce ((a * d) + (b * c)) (b * d)
 
-instance ringRatio :: Ring a => Ring (Ratio a) where
-  sub (Ratio a b) (Ratio c d) = Ratio ((a * d) - (b * c)) (b * d)
+instance ringRatio :: (Ord a, EuclideanRing a) => Ring (Ratio a) where
+  sub (Ratio a b) (Ratio c d) = reduce ((a * d) - (b * c)) (b * d)
 
-instance commutativeRingRatio :: CommutativeRing a => CommutativeRing (Ratio a)
+instance commutativeRingRatio :: (Ord a, EuclideanRing a) => CommutativeRing (Ratio a)
 
-instance euclideanRingRatio :: (CommutativeRing a, Semiring a) => EuclideanRing (Ratio a) where
-  degree _ = 1
-  div (Ratio a b) (Ratio c d) = Ratio (a * d) (b * c)
+instance euclideanRingRatio :: (Ord a, EuclideanRing a) => EuclideanRing (Ratio a) where
+  degree _ = one
+  div (Ratio a b) (Ratio c d) = reduce (a * d) (b * c)
   mod _ _ = zero
 
-instance fieldRatio :: Field a => Field (Ratio a)
+instance fieldRatio :: (Ord a, EuclideanRing a) => Field (Ratio a)
+
+reduce :: forall a. Ord a => EuclideanRing a => a -> a -> Ratio a
+reduce n d =
+  let
+    g = Prelude.gcd n d
+    d' = d / g
+  in
+    Ratio ((n / g) * signum d') (abs d')
+
+infixl 7 reduce as %
 
 numerator :: forall a. Ratio a -> a
 numerator (Ratio a _) = a
 
 denominator :: forall a. Ratio a -> a
 denominator (Ratio _ b) = b
-
-gcd :: forall a. Eq a => EuclideanRing a => Ratio a -> a
```

Comment:
Since this has been removed I think you might be able to get rid of the qualified Prelude import?